gqcp/tests/ONVBasis/OperatorString_test.cpp

```diff
@@ -0,0 +1,177 @@
+    std::vector<GQCP::Spin> spins = {GQCP::Spin::alpha, GQCP::Spin::beta, GQCP::Spin::alpha, GQCP::Spin::beta};
+
+    // Initialize the vector containing the spin components we want to use for our operator string indices.
+    // However, the dimension of this spin vector doesn't match the index vector dimension and should result in an exception being thrown.
+    std::vector<GQCP::Spin> spins_wrong = {GQCP::Spin::alpha, GQCP::Spin::beta, GQCP::Spin::alpha};
+
+    // Check the constructor.
+    BOOST_CHECK_THROW(GQCP::SpinResolvedOperatorString operator_string_wrong(indices_wrong, spins), std::invalid_argument);
+    BOOST_CHECK_THROW(GQCP::SpinResolvedOperatorString operator_string_wrong_2(indices, spins_wrong), std::invalid_argument);
+    BOOST_CHECK_NO_THROW(GQCP::SpinResolvedOperatorString operator_string_correct(indices, spins));
+}
+
+/**
+ *  Check whether the SpinUnresolvedOperatorString stores its information correctly.
+ */
+BOOST_AUTO_TEST_CASE(test_unresolved_indices) {
+
+    // Initialize the vector containing the indices we want to use for our operator string.
+    std::vector<size_t> indices = {1, 4, 7, 8};
+
+    // Initialize the SpinUnresolvedOperatorString.
+    const auto operator_string = GQCP::SpinUnresolvedOperatorString {indices};
+
+    // Get the indices from the operator string.
+    const auto operator_string_indices = operator_string.operatorIndices();
+
+    // Check whether the indices are stored correctly.
+    BOOST_CHECK_EQUAL_COLLECTIONS(indices.begin(), indices.end(), operator_string_indices.begin(), operator_string_indices.end());
+}
+
+/**
+ *  Check whether the SpinUnresolvedOperatorString`s `isZero()` check correctly checks whether the operator string will inherently equal zero when applied to any ONV.
+ */
+BOOST_AUTO_TEST_CASE(test_unresolved_isZero) {
+
+    // Initialize the vector containing the indices we want to use for our operator string.
+    // One vector does contain duplicate indices, one does not.
+    std::vector<size_t> indices_1 = {1, 4, 7, 8};
+    std::vector<size_t> indices_2 = {1, 4, 7, 1};
+
+
+    // Initialize the SpinUnresolvedOperatorStrings corresponding to the indices.
+    const auto operator_string_1 = GQCP::SpinUnresolvedOperatorString {indices_1};
+    const auto operator_string_2 = GQCP::SpinUnresolvedOperatorString {indices_2};
+
+    // Use the `isZero()` API to check whether the operator string will always result in a zero value.
+    const auto operator_string_1_zero_check = operator_string_1.isZero();
+    const auto operator_string_2_zero_check = operator_string_2.isZero();
+
+    // Compare the result of the `isZero()` check with the expected boolean value (false for the operator string without duplicate indices, true for the one with duplicate indices).
+    BOOST_CHECK_EQUAL(operator_string_1_zero_check, false);
+    BOOST_CHECK_EQUAL(operator_string_2_zero_check, true);
```

Comment:
```suggestion
    // Compare the result of the `isZero()` check with the expected boolean value (false for the operator string without duplicate indices, true for the one with duplicate indices).
    BOOST_CHECK_EQUAL( operator_string_1.isZero(), false);
    BOOST_CHECK_EQUAL(operator_string_2.isZero(), true);
```
No need to make things so verbose.